queries/vala/highlights.scm

```diff
@@ -1,297 +1,246 @@
+  (#match? @constant "^[A-Z][A-Z_0-9]*$")
+)
+
+; highlight types and probable types
+(type (symbol (_)? @namespace (identifier) @type))
+(
+  (member_access_expression . (identifier) @type)
+  (#match? @type "^[A-Z][A-Za-z_0-9]{2,}$")
+)
+
+; highlight creation methods in object creation expressions
+(
+  (object_creation_expression (type (symbol (symbol (symbol)? @include (identifier) @type) (identifier) @constructor)))
+  (#match? @constructor "^[a-z][a-z_0-9]*$")
+)
+
+(unqualified_type (symbol . (identifier) @type))
+(unqualified_type (symbol (symbol) @namespace (identifier) @type))
+
+(attribute) @attribute
+(namespace_declaration (symbol) @namespace)
+(method_declaration (symbol (symbol) @type (identifier) @function))
+(method_declaration (symbol (identifier) @function))
+(local_declaration (assignment (identifier) @variable))
+(local_function_declaration (identifier) @function)
+(destructor_declaration (identifier) @function)
+(creation_method_declaration (symbol (symbol) @type (identifier) @constructor))
+(creation_method_declaration (symbol (identifier) @constructor))
+(constructor_declaration "construct" @keyword.function)
+(enum_declaration (symbol) @type)
+(enum_value (identifier) @constant)
+(errordomain_declaration (symbol) @type)
+(errorcode (identifier) @constant)
+(constant_declaration (identifier) @constant)
+(method_call_expression (member_access_expression (identifier) @function))
+; highlight macros
+(
+ (method_call_expression (member_access_expression (identifier) @function.macro))
+ (#match? @function.macro "^assert[A-Za-z_0-9]*$" "error" "info" "debug" "print" "warning" "warning_once")
+)
+(lambda_expression (identifier) @parameter)
+(parameter (identifier) @parameter)
+(property_declaration (symbol (identifier) @property))
+(field_declaration (identifier) @field)
 [
-  "{"
-  "}"
-  "("
-  ")"
-  "["
-  "]"
-] @punctuation.bracket
-
+ (this_access)
+ (base_access)
+ (value_access)
+] @constant.builtin
+(boolean) @boolean
+(character) @character
+(escape_sequence) @character.special
+(integer) @number
+(null) @constant.builtin
+(real) @float
+(regex) @string.regex
+(string) @string
+(string_formatter) @string.special
+(template_string) @string
+(template_string_expression) @string.special
+(verbatim_string) @string
 [
-  ";"
-  ":"
-  "."
-  ","
-  "->"
-] @punctuation.delimiter
+ "var"
+ "void"
+] @type.builtin
+
+(if_directive
+  expression: (_) @preproc
+) @keyword
+(elif_directive
+  expression: (_) @preproc
+) @keyword
+(else_directive) @keyword
+(endif_directive) @keyword
 
-; Reserved keywords
+[
+ "abstract"
+ "async"
+ "break"
+ "case"
+ "catch"
+ "class"
+ "construct"
+ "continue"
+ "default"
+ "delegate"
+ "enum"
+ "errordomain"
+ "finally"
+ "get"
+ "inline"
+ "interface"
+ "namespace"
+ "new"
+ "out"
+ "override"
+ "partial"
+ "ref"
+ "set"
+ "signal"
+ "struct"
+ "switch"
+ "throw"
+ "throws"
+ "try"
+ "virtual"
+ "with"
+] @keyword
```

Comment:
I think some of those keywords fall under `@exception` or `@repeat`/`@conditional`

Reply:
Hmmm...it looks like there were some duplicate entries for some of these keywords.